server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentilesConfig.java

```diff
@@ -237,15 +243,15 @@ public InternalNumericMetricsAggregation.MultiValue createEmptyPercentileRanksAg
             DocValueFormat formatter,
             Map<String, Object> metadata
         ) {
-            return InternalTDigestPercentileRanks.empty(name, values, compression, executionHint, keyed, formatter, metadata);
+            return InternalTDigestPercentileRanks.empty(name, values, compression, getExecutionHint(), keyed, formatter, metadata);
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             super.writeTo(out);
             out.writeDouble(compression);
             if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_014)) {
-                executionHint.writeTo(out);
+                getExecutionHint().writeTo(out);
```

Comment:
Maybe use `writeOptionalWriteable(...)` here? That way we can allow exection hint to be null.

Reply:
This unfortunately means we do need to increase transport version.

Reply:
What's the benefit of leaving this as null at this point? Do we expect mutations after an object gets serialized?